fathom/src/surface.rs

```diff
@@ -85,6 +86,46 @@ pub enum Pattern<Range> {
     // RecordLiteral(Range, &'arena [((ByteRange, StringId), Pattern<'arena, Range>)]),
 }
 
+#[derive(Debug, Clone, Copy)]
+pub enum BinOp<Range> {
+    Add(Range),
+    Sub(Range),
+    Mul(Range),
+    Div(Range),
+}
+
+impl BinOp<ByteRange> {
+    fn range(&self) -> ByteRange {
+        match self {
+            BinOp::Add(range) | BinOp::Sub(range) | BinOp::Mul(range) | BinOp::Div(range) => *range,
+        }
+    }
+}
+
+impl<Range> BinOp<Range> {
+    fn as_str(&self) -> &'static str {
+        match self {
+            BinOp::Add(_) => "+",
+            BinOp::Sub(_) => "-",
+            BinOp::Mul(_) => "*",
+            BinOp::Div(_) => "/",
+        }
+    }
+
+    fn precedence(&self) -> pretty::Prec {
+        match self {
+            BinOp::Add(_) | BinOp::Sub(_) => pretty::Prec::Add,
+            BinOp::Mul(_) | BinOp::Div(_) => pretty::Prec::Mul,
+        }
+    }
```

Comment:
Not sure about this being in this module as it starts to introduce weird cycles between the modules… (perhaps something silly to worry about idk). Rust makes it easy to do that but it gives me a bit of pause. We could either pull `Prec` into here, or move this into `pretty`?